### ISSUE TITLE
Evaluation in obs_generation is super slow

### ISSUE THREAD

**Issue by tongzhoumu** [xingdi-eric-yuan/GATA-public#19]
Hi GATA authors,

I try to pre-training Graph Updater by Observation Generation. I directly used the command you provided in README, which is `python train_obs_generation.py configs/pretrain_observation_generation.yaml`.

By default, it runs evaluation on validation set every 1000 episodes. However, I found that training 1000 episodes takes ~4 mins on my machine, but the evaluation takes ~11mins, so the evaluation takes much longer time than training, which is weird to me. Is this phenomenon normal?

In addition, it will take ~45 hours to complete the the pre-training of observation generation. I am using a Tesla V100 GPU, so I feel it seems a little bit slow. How long does it take from your side?

**Comment by xingdi-eric-yuan** [comment]
Hi @tongzhoumu ,

I didn't re-run the code, but to my guess, the low speed may be caused by the fact that we compute both the negative log-likelihood loss (like people do in evaluating LM) and the F1 score between freely generated sequences with ground-truth sequences (like people do in other NLG tasks). Please see the code [here](https://github.com/xingdi-eric-yuan/GATA-public/blob/master/train_obs_generation.py#L117-L118). 

In our experiments, we observe that these two metrics align well, that said, you can probably disable the free generation computation, which I guess has used most of the eval time. 

Another thing you can do is to evaluate and save checkpoint less frequently, especially if you want to keep both metrics.

**Comment by tongzhoumu** [comment]
Thanks! The bottleneck is computing f1 score indeed. It is much faster after I commenting out the f1 score part.
